Foundation/NSUserDefaults.swift

```diff
@@ -7,46 +7,286 @@
+    }
+    public func removeObjectForKey(defaultName: String) {
+        CFPreferencesSetAppValue(defaultName._cfObject, nil, suite?._cfObject ?? kCFPreferencesCurrentApplication)
+    }
     
-    public func stringForKey(defaultName: String) -> String? { NSUnimplemented() }
-    public func arrayForKey(defaultName: String) -> [AnyObject]? { NSUnimplemented() }
-    public func dictionaryForKey(defaultName: String) -> [String : AnyObject]? { NSUnimplemented() }
-    public func dataForKey(defaultName: String) -> NSData? { NSUnimplemented() }
-    public func stringArrayForKey(defaultName: String) -> [String]? { NSUnimplemented() }
-    public func integerForKey(defaultName: String) -> Int { NSUnimplemented() }
-    public func floatForKey(defaultName: String) -> Float { NSUnimplemented() }
-    public func doubleForKey(defaultName: String) -> Double { NSUnimplemented() }
-    public func boolForKey(defaultName: String) -> Bool { NSUnimplemented() }
-    public func URLForKey(defaultName: String) -> NSURL? { NSUnimplemented() }
+    public func stringForKey(defaultName: String) -> String? {
+        guard let aVal = objectForKey(defaultName), bVal = aVal as? NSString else {
+            return nil
+        }
+        return bVal._swiftObject
+    }
+    public func arrayForKey(defaultName: String) -> [AnyObject]? {
+        guard let aVal = objectForKey(defaultName), bVal = aVal as? NSArray else {
+            return nil
+        }
+        return bVal._swiftObject
+    }
+    public func dictionaryForKey(defaultName: String) -> [String : AnyObject]? {
+        guard let aVal = objectForKey(defaultName), bVal = aVal as? NSDictionary else {
+            return nil
+        }
+        //This got out of hand fast...
+        let cVal = bVal._swiftObject
+        enum convErr: ErrorType {
+            case ConvErr
+        }
+        do {
+            let dVal = try cVal.map({ (key, val) -> (String, AnyObject) in
+                if let strKey = key as? NSString {
+                    return (strKey._swiftObject, val)
+                } else {
+                    throw convErr.ConvErr
+                }
+            })
+            var eVal = [String : AnyObject]()
+            
+            for (key, value) in dVal {
+                eVal[key] = value
+            }
+            
+            return eVal
+        } catch _ { }
+        return nil
+    }
+    public func dataForKey(defaultName: String) -> NSData? {
+        guard let aVal = objectForKey(defaultName), bVal = aVal as? NSData else {
+            return nil
+        }
+        return bVal
+    }
+    public func stringArrayForKey(defaultName: String) -> [String]? {
+        guard let aVal = objectForKey(defaultName), bVal = aVal as? NSArray else {
+            return nil
+        }
+        return (bVal._swiftObject as? [NSString])?.map({ return $0._swiftObject})
+    }
+    public func integerForKey(defaultName: String) -> Int {
```

Comment:
Wouldn't it be a better idea to return optional values in all smthngForKey functions?